File: backend/app/schemas/admin_resumen.py
from pydantic import BaseModel
from datetime import date, datetime
from typing import List, Optional


class SwapPendienteItem(BaseModel):
    id: int
    solicitante: str
    receptor: str
    fecha: Optional[date] = None
    franja_origen: Optional[str] = None
    franja_receptor: Optional[str] = None
    hace: str  # ej: "2h", "30m"


class TurnoSinConfirmarItem(BaseModel):
    id: int
    colaborador: str
    fecha: date
    franja: str


class CoberturaEnRiesgoItem(BaseModel):
    fecha: date
    franja: str
    estado: str  # "riesgo" o "critico"
    detalle: str


class ResumenSection(BaseModel):
    count: int
    items: List[dict]


class AdminResumenResponse(BaseModel):
    swaps_pendientes: ResumenSection
    turnos_sin_confirmar: ResumenSection
    cobertura_en_riesgo: ResumenSection
    colaboradores_activos: int

    class Config:
        from_attributes = True

File: frontend/src/api/admin_resumen.ts
import client from './client'

export interface SwapResumenItem {
  id: number
  solicitante: string
  receptor: string
  fecha: string | null
  franja_origen: string | null
  franja_receptor: string | null
  hace: string
}

export interface TurnoSinConfirmarItem {
  id: number
  colaborador: string
  fecha: string
  franja: string
}

export interface CoberturaRiesgoItem {
  fecha: string
  franja: string
  estado: string
  detalle: string
}

export interface ResumenSection {
  count: number
  items: any[]
}

export interface AdminResumenResponse {
  swaps_pendientes: ResumenSection
  turnos_sin_confirmar: ResumenSection
  cobertura_en_riesgo: ResumenSection
  colaboradores_activos: number
}

export const adminResumenApi = {
  getResumen: () =>
    client.get<AdminResumenResponse>('/admin/resumen'),
}
